spring-cloud-generator/src/test/java/com/google/cloud/generator/spring/composer/SpringAutoConfigClassComposerTest.java

```diff
@@ -16,55 +16,78 @@
 
 package com.google.cloud.generator.spring.composer;
 
+import static com.google.common.truth.Truth.assertThat;
+
 import com.google.api.generator.gapic.model.GapicClass;
 import com.google.api.generator.gapic.model.GapicContext;
 import com.google.api.generator.gapic.model.Service;
 import com.google.api.generator.gapic.model.Transport;
 import com.google.api.generator.test.framework.Assert;
 import com.google.api.generator.test.protoloader.GrpcRestTestProtoLoader;
+import com.google.api.generator.test.protoloader.RestTestProtoLoader;
 import com.google.api.generator.test.protoloader.TestProtoLoader;
 import org.junit.Before;
 import org.junit.Test;
 
 public class SpringAutoConfigClassComposerTest {
-  private GapicContext context;
+  private GapicContext echoContext;
+  private GapicContext echoGrpcRestContext;
+  private GapicContext echoRestContext;
   private GapicContext wickedContext;
+
   private Service echoProtoService;
+  private Service echoGrpcRestProtoService;
+  private Service echoRestProtoService;
   private Service wickedProtoService;
 
   @Before
   public void setUp() {
-    this.context = TestProtoLoader.instance().parseShowcaseEcho();
-    this.echoProtoService = this.context.services().get(0);
+    this.echoContext = TestProtoLoader.instance().parseShowcaseEcho();
+    this.echoProtoService = this.echoContext.services().get(0);
+    this.echoGrpcRestContext = GrpcRestTestProtoLoader.instance().parseShowcaseEcho();
```

Comment:
Todo: this introduction of `GrpcRestTestProtoLoader` is the source of failing checks since switching to gapic-generator-java test protoloaders in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1849/commits/f418b3e68bdff1ed4dc3b5b3e8b838bb08f7e918.

Options to fix:
1.  Switch `GRPC_REST` tests back to using `TestProtoLoader` and changing transport in parsed context
2. Copy over [`echo_grpcrest.proto`](https://github.com/googleapis/sdk-platform-java/blob/main/gapic-generator-java/src/test/proto/echo_grpcrest.proto), which seem to be what is used by `GrpcRestTestProtoLoader`.

Reply:
Does the second option mean we have to keep an identical copy of `echo_grpcrest.proto` in spring-cloud-gcp? If that's the case, I would prefer the first option.

Reply:
Agreed - I've updated the setup with the first option in https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/1849/commits/0ff01b2a11f741c35f72a8b749c5d42e3137a119.

Trying to wrap my head around the variations of echo proto handling - I'm guessing there are different test proto loaders set up by transport in `gapic-generator-java` for use in its golden tests for transport-specific configs/implementations? So far spring composer tests using the echo proto will just need the `Transport` types to differ accordingly in the parsed `GapicContext`, so this setup is sufficient. In the future, if more test proto parsing logic start to diverge from gapic-generator-java's setup, we can consider placing them into a new `SpringTestProtoLoader` (maintained here).